### ISSUE TITLE
World difficulty breaks when using /spawn and /home commands to travel between overworld and the void

### ISSUE THREAD

**Issue by ToggleLifeOff** [Nomi-CEu/Nomi-CEu#297]
**Pack Version:**
1.5.2

**Launcher:**
Curse Forge

**Mode:**
Normal Mode

**Environment:**
Singleplayer, no extra mods induce this issue, the only mod added was optifine

**Description:**
When you traverse from overworld to the void with /home command the world difficulty changes randomly from peaceful to normal, and even when you turn it back to peaceful the mobs still spawn, also the hunger bar goes down.

**Expected Behavior:**
The difficulty should not change, and the mobs should not spawn, when the difficulty is set to peaceful.

**Steps To Reproduce:**
1. create a new world
2. change the difficulty to peaceful
3. get a void cake and teleport there to set up a /home void command
4. teleport with /spawn back to overworld
5. use /home void to go back and check the difficulty, if it has changed turn it back to peaceful
6. watch the hunger bar, and create a dark spot to test if mobs are spawning (it is a bit random, so try copule times)
![github bug report 1](https://user-images.githubusercontent.com/87280942/226722375-d543377b-5be4-46be-8994-8597b65c191a.png)
![github bug report 2](https://user-images.githubusercontent.com/87280942/226722376-0110e45f-688a-497b-8f12-12d3f0f5fa5c.png)

**Crash Log:**
no crash occurs when this bug happens

**Screenshots:**

As you can see on the second screenshot, peaceful is on, but there is clearly a creeper next to me 🙃

**Comment by IntegerLimit** [comment]
Um... I'll have a look at this. This seems like a problem with Global Gamerules (which was removed) though. Did you happen to press `update` on your CurseForge Launcher from something like 1.4.3 or 1.5? Please try this on another launcher, such as Prism Launcher.

**Comment by ToggleLifeOff** [comment]
well, this happened on a commpletely new instance of nomifactory ceu. I will definitely test this on other launchers and try to come up with different conditions that could/or not replicate this. Also I noticed this bug doesn't happen when you use the cakes, tho. but refilling them is just a pain in the bun for me so... like they say laziness is the mother of invention

**Comment by IntegerLimit** [comment]
Huh, I'll have a look at it. Thanks.

**Comment by ToggleLifeOff** [comment]
Hey, coming back I have tested this on Prism, and it shows the same behaviour. Once I use the /home void command, the console log just says "difficulty changed to NORMAL from PEACEFUL". After reading a bit and testing with multiple version of FTB Utilities I can't tell if this problem lays on the mod side, or the client side.

**Comment by IntegerLimit** [comment]
Ok, thanks, can you send the log?

**Comment by 0xC4DE** [comment]
This sounds a whole lot like a problem with the world difficulty locker config file. Since it is supposed to lock and force normal mode within a world that is in normal mode.

I know it is minecraft behavior to make difficulty per-dimension instead of per-world. Whereas if you do /difficulty peaceful, it may change the difficulty back to the world default (in this case it would be normal)

Have you changed the world difficulty config?

**Comment by Joshobo** [comment]
As _0xC4DE_ mentioned it seems like an issue with the difficultylock.cfg.

To get minecraft/nomi to respect the difficulty I changed to in-game (peaceful) using the options menu, I have had to modify the difficultylock.cfg and set both the forcespeaceul=true and shouldlockdifficulty=true

In **expert** mode this isn't an issue as they are set to peaceful and locked.
In **normal** every option is false and I assume difficulty lock then defaults back to minecrafts defaults of normal difficulty.

When you use teleport commands this seems to force reload the difficulty lock defaults for that dimension. Oddly i don't remember it happening when you use Journey Map to teleport between dimensions.
